src/main/python/apache/aurora/config/loader.py

```diff
@@ -56,17 +62,49 @@ def flush_schemas(cls):
     cls.register_schema(base_schema)
 
   @classmethod
-  def load(cls, loadable):
-    return cls.load_raw(loadable).environment
+  def gen_content_key(cls, loadable):
+    key = None
+    if isinstance(loadable, Compatibility.stringy) and os.path.isfile(loadable):
+      with open(loadable) as fp:
+        key = hashlib.md5(fp.read()).hexdigest()
+    elif hasattr(loadable, 'read') and callable(loadable.read):
+      key = hashlib.md5(loadable.read()).hexdigest()
+      loadable.seek(0)
+    return key
```

Comment:
We will not cache results if `loadable` is from a [Resource](https://github.com/wickman/pystachio/blob/master/pystachio/config.py#L88). Should we use the `loadable` as the key instead?

Possible improvement - wonder if we should use the different [ConfigExecutors](https://github.com/wickman/pystachio/blob/master/pystachio/config.py#L148) and call the `matches` function instead of replicating the logic?

Reply:
We could use the loadable as the key (that was the previous implementation) however someone could change the loadable's content and the cache would not be updated.

I used the ConfigExecutors.matches instead of duplicating the logic.

Reply:
The documentation makes things much clearer. LGTM.